### DISCUSSION TITLE
Raise for respone content type

### DISCUSSION THREAD

**Discussion by 4bo** [encode/httpx#1723] [General] [answered]
I have many links to download, but many of them is not of desired content type. i.e. I'm just interested in ```text/html```, but prior to actually ```get``` them, the content type is not known.
how can i check content type before completely downloading it. i could perform a ```head``` request, bot it is not guaranteed that target servers support head request.

**Comment by johtso** [comment, marked as answer]
You could just make your requests in streaming mode and not read the content. https://www.python-httpx.org/quickstart/#streaming-responses

The last example in that section of the docs is basically your use case:

```python
>>> with httpx.stream("GET", "https://www.example.com") as r:
...     if r.headers['Content-Length'] < TOO_LONG:
...         r.read()
...         print(r.text)
```
